### ISSUE TITLE
Error running dockertest after upgrading to Mac Monterey

### ISSUE THREAD

**Issue by jmcervera** [ory/dockertest#320]
Hi, I've just have a crash when running my tests.

The crash occurs on the pool.BuildAndRun statement

```
	pool, err := dockertest.NewPool("")
	if err != nil {
		logrus.Fatalf("Could not connect to docker: %s", err)
	}
	resource, err := pool.BuildAndRun("mydb", dockerfile, []string{})
```
This is the dump and the error I got on the console.
I'm sorry to leave this output here, but it might be helpful to you

`
objc[17776]: Attempt to use unknown class 0x7ff85e55b360.
SIGABRT: abort
PC=0x7ff81cd27dfe m=3 sigcode=0

goroutine 0 [idle]:
runtime: unknown pc 0x7ff81cd27dfe
stack: frame={sp:0x70000ab40e68, fp:0x0} stack=[0x70000aac2350,0x70000ab41f50)
000070000ab40d68:  0000000000000002  0000000000000002 
000070000ab40d78:  0000000002788000  0000000002788400 
000070000ab40d88:  000000000000000a  000070000ab40df0 
000070000ab40d98:  00007ff81cb5d639  0000000000000000 
000070000ab40da8:  f69b4000c000100a  000000000000ffff 
000070000ab40db8:  0000000000000030  0000000000000000 
000070000ab40dc8:  000000000000002d  000000000000002d 
000070000ab40dd8:  0000000000000000  0000000002788000 
000070000ab40de8:  00007ff85e56f000  000070000ab40e20 
000070000ab40df8:  00007ff81cb5d566  00007ff81cb5d52e 
000070000ab40e08:  000000000000002d  0000000000000001 
000070000ab40e18:  00007ff85e56f000  000070000ab40e60 
000070000ab40e28:  00007ff81cb79ad6  000000000000002c 
000070000ab40e38:  000000000000002d  0000600003300000 
000070000ab40e48:  0000000000000003  0000600002c00000 
000070000ab40e58:  0000000000000000  000070000ab40e90 
000070000ab40e68: <00007ff81cd29893  0000000000000000 
000070000ab40e78:  000000200ab40ed0  0000600002c00000 
000070000ab40e88:  000070000ab40ed0  0000600003300000 
000070000ab40e98:  0000000000000003  0000000000000080 
000070000ab40ea8:  0000000000000001  000070000ab40ec0 
000070000ab40eb8:  00007ff81cd29843  000070000ab40f00 
000070000ab40ec8:  00007ff81cbf5bce  0000600002c00080 
000070000ab40ed8:  0000000000000000  0000600002c00000 
000070000ab40ee8:  00007ffffffffff8  00007ff85e55b360 
000070000ab40ef8:  0000000080080001  000070000ab40fe0 
000070000ab40f08:  00007ff81cbf5b5c  0000003000000010 
000070000ab40f18:  000070000ab40ff0  000070000ab40f30 
000070000ab40f28:  0000000000000000  0000000000000000 
000070000ab40f38:  00007ff85e55b360  00007ff85e55b360 
000070000ab40f48:  0000000000000001  00000000ffffffff 
000070000ab40f58:  0000000000000000  0000000000000000 
runtime: unknown pc 0x7ff81cd27dfe
stack: frame={sp:0x70000ab40e68, fp:0x0} stack=[0x70000aac2350,0x70000ab41f50)
000070000ab40d68:  0000000000000002  0000000000000002 
000070000ab40d78:  0000000002788000  0000000002788400 
000070000ab40d88:  000000000000000a  000070000ab40df0 
000070000ab40d98:  00007ff81cb5d639  0000000000000000 
000070000ab40da8:  f69b4000c000100a  000000000000ffff 
000070000ab40db8:  0000000000000030  0000000000000000 
000070000ab40dc8:  000000000000002d  000000000000002d 
000070000ab40dd8:  0000000000000000  0000000002788000 
000070000ab40de8:  00007ff85e56f000  000070000ab40e20 
000070000ab40df8:  00007ff81cb5d566  00007ff81cb5d52e 
000070000ab40e08:  000000000000002d  0000000000000001 
000070000ab40e18:  00007ff85e56f000  000070000ab40e60 
000070000ab40e28:  00007ff81cb79ad6  000000000000002c 
000070000ab40e38:  000000000000002d  0000600003300000 
000070000ab40e48:  0000000000000003  0000600002c00000 
000070000ab40e58:  0000000000000000  000070000ab40e90 
000070000ab40e68: <00007ff81cd29893  0000000000000000 
000070000ab40e78:  000000200ab40ed0  0000600002c00000 
000070000ab40e88:  000070000ab40ed0  0000600003300000 
000070000ab40e98:  0000000000000003  0000000000000080 
000070000ab40ea8:  0000000000000001  000070000ab40ec0 
000070000ab40eb8:  00007ff81cd29843  000070000ab40f00 
000070000ab40ec8:  00007ff81cbf5bce  0000600002c00080 
000070000ab40ed8:  0000000000000000  0000600002c00000 
000070000ab40ee8:  00007ffffffffff8  00007ff85e55b360 
000070000ab40ef8:  0000000080080001  000070000ab40fe0 
000070000ab40f08:  00007ff81cbf5b5c  0000003000000010 
000070000ab40f18:  000070000ab40ff0  000070000ab40f30 
000070000ab40f28:  0000000000000000  0000000000000000 
000070000ab40f38:  00007ff85e55b360  00007ff85e55b360 
000070000ab40f48:  0000000000000001  00000000ffffffff 
000070000ab40f58:  0000000000000000  0000000000000000 

goroutine 34 [syscall]:
runtime.cgocall(0x10018c0, 0xc000134f70, 0xc000010020)
        /Users/juanmacervera/.go/src/runtime/cgocall.go:154 +0x7a fp=0xc000134f40 sp=0xc000134f08 pc=0x104503a
os/user._Cfunc_mygetpwuid_r(0x1f5, 0xc00030e000, 0x280a000, 0x1000, 0xc000010020, 0x0)
        _cgo_gotypes.go:181 +0x6b fp=0xc000134f70 sp=0xc000134f40 pc=0x16d2bcb
os/user.lookupUnixUid.func1.1(0x1f5, 0xc00030e000, 0xc000072030, 0xc000010020, 0xc000060900)
        /Users/juanmacervera/.go/src/os/user/cgo_lookup_unix.go:103 +0x105 fp=0xc000134fc8 sp=0xc000134f70 pc=0x16d46c5
os/user.lookupUnixUid.func1(0x16d32b1)
        /Users/juanmacervera/.go/src/os/user/cgo_lookup_unix.go:103 +0x5d fp=0xc000135010 sp=0xc000134fc8 pc=0x16d475d
os/user.retryWithBuffer(0xc000072030, 0xc000135158, 0x0, 0x0)
        /Users/juanmacervera/.go/src/os/user/cgo_lookup_unix.go:247 +0x4f fp=0xc000135080 sp=0xc000135010 pc=0x16d436f
os/user.lookupUnixUid(0x1f5, 0x0, 0x0, 0x0)
        /Users/juanmacervera/.go/src/os/user/cgo_lookup_unix.go:96 +0x1d1 fp=0xc000135190 sp=0xc000135080 pc=0x16d32b1
os/user.current(0x10c67ea, 0xc000135208, 0x10c60ca)
        /Users/juanmacervera/.go/src/os/user/cgo_lookup_unix.go:49 +0x35 fp=0xc0001351d8 sp=0xc000135190 pc=0x16d2f75
os/user.Current.func1()
        /Users/juanmacervera/.go/src/os/user/lookup.go:15 +0x34 fp=0xc000135218 sp=0xc0001351d8 pc=0x16d44f4
sync.(*Once).doSlow(0x1c532a0, 0x18a4118)
        /Users/juanmacervera/.go/src/sync/once.go:68 +0x10a fp=0xc0001352e8 sp=0xc000135218 pc=0x10c67ea
sync.(*Once).Do(0x1c532a0, 0x18a4118)
        /Users/juanmacervera/.go/src/sync/once.go:59 +0x69 fp=0xc000135308 sp=0xc0001352e8 pc=0x10c66c9
os/user.Current(0x16dac55, 0x0, 0x0)
        /Users/juanmacervera/.go/src/os/user/lookup.go:15 +0x4b fp=0xc000135350 sp=0xc000135308 pc=0x16d252b
os/user.LookupId(0xc000024109, 0x3, 0xc000024109, 0x3, 0x0)
        /Users/juanmacervera/.go/src/os/user/lookup.go:42 +0x34 fp=0xc0001353b0 sp=0xc000135350 pc=0x16d26b4
archive/tar.statUnix(0x19468f8, 0xc000308270, 0xc00030c000, 0x1c53800, 0xc00002e360)
        /Users/juanmacervera/.go/src/archive/tar/stat_unix.go:39 +0x8b5 fp=0xc000135498 sp=0xc0001353b0 pc=0x16dac55
archive/tar.FileInfoHeader(0x19468f8, 0xc000308270, 0x0, 0x0, 0xc000308270, 0xc00006f6c0, 0x11182ec)
        /Users/juanmacervera/.go/src/archive/tar/common.go:702 +0x888 fp=0xc000135638 sp=0xc000135498 pc=0x16d82e8
github.com/ory/dockertest/v3/docker/pkg/archive.FileInfoHeader(0xc00002e2f0, 0xa, 0x19468f8, 0xc000308270, 0x0, 0x0, 0xc000072020, 0xc00006f738, 0x109442c)
        /Users/juanmacervera/go/pkg/mod/github.com/ory/dockertest/v3@v3.8.0/docker/pkg/archive/archive.go:359 +0x7d fp=0xc0001356d0 sp=0xc000135638 pc=0x16ea53d
github.com/ory/dockertest/v3/docker/pkg/archive.(*tarAppender).addTarFile(0xc0000200c0, 0xc00002e2a0, 0x5a, 0xc00002e2f0, 0xa, 0x0, 0x0)
        /Users/juanmacervera/go/pkg/mod/github.com/ory/dockertest/v3@v3.8.0/docker/pkg/archive/archive.go:473 +0x10c fp=0xc000135890 sp=0xc0001356d0 pc=0x16eaf2c
github.com/ory/dockertest/v3/docker/pkg/archive.TarWithOptions.func1.2(0xc00002e2a0, 0x5a, 0x19468f8, 0xc0003081a0, 0x0, 0x0, 0x0, 0x0)
        /Users/juanmacervera/go/pkg/mod/github.com/ory/dockertest/v3@v3.8.0/docker/pkg/archive/archive.go:883 +0x84c fp=0xc000135a78 sp=0xc000135890 pc=0x16ed12c
path/filepath.walk(0xc00002e2a0, 0x5a, 0x19468f8, 0xc0003081a0, 0xc00006fec0, 0x0, 0x0)
        /Users/juanmacervera/.go/src/path/filepath/path.go:414 +0x54c fp=0xc000135b60 sp=0xc000135a78 pc=0x14284ac
path/filepath.walk(0xc00002e0c0, 0x51, 0x19468f8, 0xc0003080d0, 0xc000135ec0, 0x0, 0x3)
        /Users/juanmacervera/.go/src/path/filepath/path.go:438 +0x3c5 fp=0xc000135c48 sp=0xc000135b60 pc=0x1428325
path/filepath.Walk(0xc00002e0c0, 0x51, 0xc00006fec0, 0x19317d8, 0x1)
        /Users/juanmacervera/.go/src/path/filepath/path.go:501 +0x15b fp=0xc000135ca8 sp=0xc000135c48 pc=0x142867b
github.com/ory/dockertest/v3/docker/pkg/archive.TarWithOptions.func1(0xc0002a8000, 0x1940428, 0xc000292090, 0xc000290090, 0xc000282160, 0xc000284120)
        /Users/juanmacervera/go/pkg/mod/github.com/ory/dockertest/v3@v3.8.0/docker/pkg/archive/archive.go:801 +0x67e fp=0xc000135fb0 sp=0xc000135ca8 pc=0x16edc9e
runtime.goexit()
        /Users/juanmacervera/.go/src/runtime/asm_amd64.s:1371 +0x1 fp=0xc000135fb8 sp=0xc000135fb0 pc=0x10b5861
created by github.com/ory/dockertest/v3/docker/pkg/archive.TarWithOptions
        /Users/juanmacervera/go/pkg/mod/github.com/ory/dockertest/v3@v3.8.0/docker/pkg/archive/archive.go:743 +0x415

goroutine 1 [select]:
io.(*pipe).Read(0xc0002801e0, 0xc0002c6000, 0x8000, 0x8000, 0xc00021e540, 0x109442c, 0x8000)
        /Users/juanmacervera/.go/src/io/pipe.go:57 +0x18e
io.(*PipeReader).Read(0xc000290088, 0xc0002c6000, 0x8000, 0x8000, 0x0, 0x0, 0x2)
        /Users/juanmacervera/.go/src/io/pipe.go:134 +0x6e
io.copyBuffer(0xa3b9088, 0xc000282260, 0x193bb80, 0xc000290088, 0xc0002c6000, 0x8000, 0x8000, 0x0, 0xa3b9088, 0xc00021e670)
        /Users/juanmacervera/.go/src/io/io.go:423 +0x14c
io.Copy(...)
        /Users/juanmacervera/.go/src/io/io.go:382
net/http.(*transferWriter).doBodyCopy(0xc0002a80a0, 0xa3b9088, 0xc000282260, 0x193bb80, 0xc000290088, 0x1094d5c, 0xc0002c50cb, 0x1875ab7)
        /Users/juanmacervera/.go/src/net/http/transfer.go:409 +0x7c
net/http.(*transferWriter).writeBody(0xc0002a80a0, 0x193bf00, 0xc0002900a0, 0x0, 0x0)
        /Users/juanmacervera/.go/src/net/http/transfer.go:356 +0x789
net/http.(*Request).write(0xc0002b8000, 0x193b520, 0xc0002880c0, 0xa3b9000, 0x0, 0x0, 0x0, 0x0)
        /Users/juanmacervera/.go/src/net/http/request.go:697 +0x911
net/http.(*Request).Write(...)
        /Users/juanmacervera/.go/src/net/http/request.go:527
github.com/ory/dockertest/v3/docker.(*Client).stream(0xc000148370, 0x1875fe8, 0x4, 0xc0002b6030, 0x27, 0x1, 0xc00028a000, 0x193bb80, 0xc000290088, 0x193c8a0, ...)
        /Users/juanmacervera/go/pkg/mod/github.com/ory/dockertest/v3@v3.8.0/docker/client.go:580 +0x8ac
github.com/ory/dockertest/v3/docker.(*Client).BuildImage(0xc000148370, 0x1877d7a, 0x8, 0x188d94c, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /Users/juanmacervera/go/pkg/mod/github.com/ory/dockertest/v3@v3.8.0/docker/image.go:551 +0x8b8
github.com/ory/dockertest/v3.(*Pool).BuildAndRunWithBuildOptions(0xc0002386d0, 0xc00021fb20, 0xc00021fc00, 0x0, 0x0, 0x0, 0x1707006, 0xc0002386d0, 0x10)
        /Users/juanmacervera/go/pkg/mod/github.com/ory/dockertest/v3@v3.8.0/dockertest.go:317 +0x1fc
github.com/ory/dockertest/v3.(*Pool).BuildAndRunWithOptions(0xc0002386d0, 0x188d928, 0x2e, 0xc00021fc00, 0x0, 0x0, 0x0, 0x1f00000000000000, 0x1f0000c000133380, 0xc00021fbe0)
        /Users/juanmacervera/go/pkg/mod/github.com/ory/dockertest/v3@v3.8.0/dockertest.go:340 +0x113
github.com/ory/dockertest/v3.(*Pool).BuildAndRun(...)
        /Users/juanmacervera/go/pkg/mod/github.com/ory/dockertest/v3@v3.8.0/dockertest.go:345

`

**Comment by jmcervera** [comment]
I just wanted to give a little more information about the error.
I'm making my tests building and starting just a myql docker image, 
everything went fine before updating to macOS  Monterey, 
I have siscovered that my tests are still running well when I run them without th `-race` optio.
`go test ./...` works
but `go test -race ./...` produces the error I have uploaded. 
This happen in all my projects.
I hope this gives you a clue of something that could be generating the errors
